video/out/opengl/context_vdpau.c

```diff
@@ -280,110 +284,127 @@ static int glx_init(struct MPGLContext *ctx, int flags)
+static bool glx_reconfig(struct ra_ctx *ctx)
 {
     vo_x11_config_vo_window(ctx->vo);
-    return 0;
+    return true;
 }
 
-static int glx_control(struct MPGLContext *ctx, int *events, int request,
-                       void *arg)
+static int glx_control(struct ra_ctx *ctx, int *events, int request, void *arg)
 {
     return vo_x11_control(ctx->vo, events, request, arg);
 }
 
-static void glx_start_frame(struct MPGLContext *ctx)
+static struct ra_tex *glx_start_frame(struct ra_ctx *ctx)
 {
     struct priv *p = ctx->priv;
-    struct vdp_functions *vdp = &p->vdp->vdp;
-    VdpStatus vdp_st;
-    GL *gl = ctx->gl;
-
-    struct surface *surface = &p->surfaces[p->current_surface];
-
-    if (surface->surface != VDP_INVALID_HANDLE) {
-        VdpTime prev_vsync_time;
-        vdp_st = vdp->presentation_queue_block_until_surface_idle(p->vdp_queue,
-                                                                  surface->surface,
-                                                                  &prev_vsync_time);
-        CHECK_VDP_WARNING(ctx, "waiting for surface failed");
+    GL *gl = &p->gl;
+
+    struct surface *surf = &p->surfaces[p->idx_surfaces];
+    if (surf->w != ctx->vo->dwidth || surf->h != ctx->vo->dheight ||
+        surf->surface == VDP_INVALID_HANDLE)
+    {
+        if (!recreate_vdpau_surface(ctx, surf))
+            return NULL;
     }
 
-    if (surface->w != ctx->vo->dwidth || surface->h != ctx->vo->dheight)
-        recreate_vdpau_surface(ctx, surface);
-
-
-    ctx->main_fb = surface->fbo; // 0 if creating the surface failed
+    assert(!surf->mapped);
+    gl->VDPAUMapSurfacesNV(1, &surf->registered);
+    surf->mapped = true;
 
-    if (surface->surface != VDP_INVALID_HANDLE) {
-        gl->VDPAUMapSurfacesNV(1, &surface->registered);
-        surface->mapped = true;
-    }
+    ra_gl_ctx_resize(ctx, surf->w, surf->h, surf->fbo);
+    return ra_gl_ctx_start_frame(ctx);
 }
 
-static void glx_swap_buffers(struct MPGLContext *ctx)
+static bool glx_submit_frame(struct ra_ctx *ctx, const struct vo_frame *frame)
 {
     struct priv *p = ctx->priv;
+    GL *gl = &p->gl;
     struct vdp_functions *vdp = &p->vdp->vdp;
     VdpStatus vdp_st;
-    GL *gl = ctx->gl;
 
-    struct surface *surface = &p->surfaces[p->current_surface];
-    if (surface->surface == VDP_INVALID_HANDLE)
-        return; // surface alloc probably failed before
-
-    if (surface->mapped)
-        gl->VDPAUUnmapSurfacesNV(1, &surface->registered);
-    surface->mapped = false;
+    struct surface *surf = &p->surfaces[p->idx_surfaces];
+    assert(surf->surface != VDP_INVALID_HANDLE);
+    assert(surf->mapped);
+    gl->VDPAUUnmapSurfacesNV(1, &surf->registered);
+    surf->mapped = false;
 
-    vdp_st = vdp->presentation_queue_display(p->vdp_queue, surface->surface,
-                                             0, 0, 0);
+    vdp_st = vdp->presentation_queue_display(p->vdp_queue, surf->surface, 0, 0, 0);
```

Comment:
What's even the point of surface -> surf?

Reply:
80 col in some other places. I don't like typing that much, and I basically rewrite most of this code (even if it ends up looking similar)

Reply:
Only complaining about how it inflates the diff and makes it harder to read.